tests/components/samsungtv/test_media_player.py

```diff
@@ -56,254 +132,429 @@ class UnhandledResponse(Exception):
+    """Testing unhandled response exception."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    remote.control = mock.Mock(side_effect=exceptions.UnhandledResponse("Boom"))
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_VOLUME_UP, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == STATE_ON
+
+
+async def test_send_key_os_error(hass, remote):
+    """Testing broken pipe Exception."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    remote.control = mock.Mock(side_effect=OSError("Boom"))
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_VOLUME_UP, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == STATE_OFF
+
+
+async def test_name(hass, remote):
+    """Test for name property."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    state = hass.states.get(ENTITY_ID)
+    assert state.attributes[ATTR_FRIENDLY_NAME] == "fake"
+
+
+async def test_state_with_mac(hass, remote, wakeonlan):
+    """Test for state property."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == STATE_ON
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_OFF, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == STATE_OFF
+
+
+async def test_state_without_mac(hass, remote):
+    """Test for state property."""
+    await setup_samsungtv(hass, MOCK_CONFIG_NOMAC)
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_VOLUME_UP, {ATTR_ENTITY_ID: ENTITY_ID_NOMAC}, True
+    )
+    state = hass.states.get(ENTITY_ID_NOMAC)
+    assert state.state == STATE_ON
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_OFF, {ATTR_ENTITY_ID: ENTITY_ID_NOMAC}, True
+    )
+    state = hass.states.get(ENTITY_ID_NOMAC)
+    assert state.state == STATE_OFF
+
+
+async def test_supported_features_with_mac(hass, remote):
+    """Test for supported_features property."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    state = hass.states.get(ENTITY_ID)
+    assert (
+        SUPPORT_SAMSUNGTV | SUPPORT_TURN_ON == state.attributes[ATTR_SUPPORTED_FEATURES]
+    )
+
+
+async def test_supported_features_without_mac(hass, remote):
+    """Test for supported_features property."""
+    await setup_samsungtv(hass, MOCK_CONFIG_NOMAC)
+    state = hass.states.get(ENTITY_ID_NOMAC)
+    assert SUPPORT_SAMSUNGTV == state.attributes[ATTR_SUPPORTED_FEATURES]
+
+
+async def test_device_class(hass, remote):
+    """Test for device_class property."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    entity = hass.data[DOMAIN].get_entity(ENTITY_ID)
+    assert entity
+    assert entity.device_class == DEVICE_CLASS_TV
```

Comment:
The device class is available on the state as a state attribute, `ATTR_DEVICE_CLASS`. Please use the entity state to assert the entity device class.